openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/LineageRepository.java

```diff
@@ -140,86 +161,233 @@ public void addLineage(AddLineage addLineage) {
             Relationship.UPSTREAM.ordinal(),
             detailsJson);
     addLineageToSearch(from, to, addLineage.getEdge().getLineageDetails());
+
+    // build Extended Lineage
+    buildExtendedLineage(from, to, lineageDetails, relationAlreadyExists);
+  }
+
+  private void buildExtendedLineage(
+      EntityReference from,
+      EntityReference to,
+      LineageDetails lineageDetails,
+      boolean childRelationExists) {
+    boolean addService =
+        Entity.entityHasField(from.getType(), "service")
```

Comment:
we need to add some constants for these strings

Reply:
sure